### DISCUSSION TITLE
JSON Serialization

### DISCUSSION THREAD

**Discussion by dylanjaide** [scikit-hep/hist#351] [Ideas]
We would like to be able to transfer `hist` objects using an API (use case: [LHCb Analysis Productions](https://gitlab.cern.ch/lhcb-dpa/analysis-productions/lbapcommon/-/merge_requests/39)), which will require conversion to JSON and back. If I understand correctly, there isn't yet a standardised way of doing this for `hist` histograms. If not, is this something that could be added?

**Comment by henryiii** [comment]
We've been thinking about this for a while, but haven't worked on a standard API yet. It would be nice to define a standard format that multiple packages could use, and ideally would be supported by Boost.Histogram as well as boost-histogram (possibly as an optional component). We could even document it in UHI, for example.

I can work on this after December 8 or so. CC @HDembinski and @jpivarski, thoughts? Can we start working on a draft?

**Reply by eduardo-rodrigues** [reply]
Hi all. To put more weight/interest in this: Hist, boost-histogram and others are now dependencies for the LHCb offline analysis stack and in particular analysis productions in this instance, which is the centralised framework with which we create ntuples to users on skimmed and trimmed data. The framework is part of Run-3 preparations but is already in use with legacy (Runs 1 and 2) data :-).

**Comment by DraTeots** [comment]
Hi All, we are also interested in this feature. The idea is to create histograms and save to JSon to work with them on JS frontend. We could do this easily as a custom format but... We use it for EIC physics group analysis done by multiple physicists and students and having a standard functions and format specification would be much (MUCH) better than hanging our homebrew serializer around.

**Reply by henryiii** [reply]
The spec is in uhi, and there will be a release quite soon with the feature. I've released boost-histogram 1.6.0 and then 1.6.1 (earlier today) with the backend support. A Hist and UHI release will be coming soon. Feel free to look over and comment on the spec at https://uhi.readthedocs.io/en/latest/serialization.html.

**Reply by DraTeots** [reply]
Thank you very much! This is excellent. Even if it takes time to release, we can use the schema and prepare the frontend.
